.github/workflows/deploy-to-vcr-on-commit.yml

```diff
@@ -0,0 +1,58 @@
+name: Deploy to VCR on PR
+
+on:
+  pull_request:
+    types: [opened, labeled]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+# avoid running more than one workflow per PR in repo
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-and-test:
+    runs-on: [vcp-runner]
+    if: github.event.action == 'opened' || github.event.label.name == 'vcr-deploy'
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.4
+          cache: npm
+
+      - name: Install VCR CLI
+        run: |
+          sudo curl -L https://raw.githubusercontent.com/Vonage/cloud-runtime-cli/main/script/install.sh | sudo sh
+          vcr -v
+
+      - name: Replace some values in the manifest so that it does not break other deployments and is unique
+        run: |
+          cp vcr-gha.yml vcr-gha.temp.yml
+          SHORT_SHA=$(echo "${GITHUB_SHA}" | cut -c1-7)
+          sed -i "s/name: vonage-video-react-app/name: vonage-video-react-app-$SHORT_SHA/g" vcr-gha.temp.yml
+          sed -i 's/<DOMAIN>//g' vcr-gha.temp.yml
+
+      - name: Deploy vcr
+        id: deploy
+        run: |
+          vcr deploy --filename vcr-gha.temp.yml --app-id ${{ secrets.APP_ID }} --api-key ${{ secrets.VCR_API_KEY }} --api-secret ${{ secrets.VCR_API_SECRET }} --region aws.euw1 --graphql-endpoint https://graphql.euw1.runtime.vonage.cloud/v1/graphql --timeout=15m 2>&1 | tee deploy-vcr-logs.log
+          VCR_URL=$(grep -oE 'https://[a-zA-Z0-9.-]+\.vonage\.cloud' deploy-vcr-logs.log | head -1)
+          echo "vcr_url=$VCR_URL" >> $GITHUB_OUTPUT
+
+      - name: Comment the VCR instance link on PR
+        uses: peter-evans/create-or-update-comment@v3
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            🚀 VCR instance deployed for commit hash: `${{ github.sha }}`)
```

Comment:
What if a deploy goes side ways?  Do we revert to the last know good deploy?  Do we get notified as well, in both cases?

Reply:
the job will just fail and we all going to have a big sad 😭 it won't post a message though and the workflow will fail

Reply:
If we're going to do continuous deployment (CD), then we need better observability, please.  If an auto-deployment fails, then the team needs to be notified somehow, especially since the app heavily used internally.

It was be *sweeeet* if we could also fallback on a previous release too, if an auto-update fails.  This can be done in a follow up PR, though.

Alternatively, maybe GHA has a plugin that gives CD for us for free 🤷‍♂️.